app/models/course/assessment/category.rb

```diff
@@ -1,6 +1,31 @@
 # Represents a category of assessments. This is typically 'Mission' and 'Training'.
 class Course::Assessment::Category < ActiveRecord::Base
   belongs_to :course, inverse_of: :assessment_categories
-  has_many :tabs, class_name: Course::Assessment::Tab.name, inverse_of: :category
+  has_many :tabs, class_name: Course::Assessment::Tab.name,
+           inverse_of: :category, dependent: :destroy
   has_many :assessments, through: :tabs
+
+  accepts_nested_attributes_for :tabs
+
+  default_scope { order(:weight) }
+
+  before_destroy :before_destroy_validation
+
+  # Returns a boolean value indicating if this category can be safely destroyed.
+  # This is the case when another category with at least one tab exists.
+  #
+  # @return [Boolean]
+  def safe_to_destroy?
```

Comment:
Any suggestions for the name? `final_category?` (negated) maybe? Tabs also have this and call the category version as part of their `before_destroy`.